scripts/generate-changelog.sh

```diff
@@ -1,31 +1,47 @@
 # /bin/bash
 
-# Usage: ./scripts/generate-changelog.sh v1.2 "April 01, 2021"
+# Usage: ./scripts/generate-changelog.sh v1.2 2021-04-01
 # or: ./scripts/generate-changelog.sh vUNSTABLE
 
 set -e
 
 VERSION="$1"
 DATE="$2"
 
-cd `dirname $0`/../changelogs
-
-# Pre-cleanup just in case it wasn't done on the last run
-rm -f rendered.md
-
-# Generate changelog
-towncrier --yes
+if [ -z "$VERSION" ]; then
+    echo "ERROR: The version of the changelog must be provided"
+    exit 1
+fi
 
 if [ "$VERSION" = "vUNSTABLE" ]; then
     TITLE="Changes since last release"
     LINKTITLE="Unstable"
     FILENAME="unstable.md"
+    DATE=$(date -Iseconds)
 else
     TITLE="$VERSION Changelog"
     LINKTITLE="$VERSION"
     FILENAME="$VERSION.md"
+
+    if [ -z "$DATE" ]; then
```

Comment:
Note: We could actually always get the date from the system instead of providing it as a parameter of the script, if we assume that the changelog will always be created on the date of the release.

Even if that's not the case, the date could be edited manually in the changelog page after it is generated.

Reply:
Speaking as a maintainer of Synapse, [our release script](https://github.com/element-hq/synapse/blob/develop/scripts-dev/release.py) uses the current date and it's perfectly serviceable. It'd be inconvenient (and prone to human error) to have to type it in manually each time.

So +1 to automating that.

Reply:
Done